source/contracts/architecture.md

```diff
@@ -0,0 +1,108 @@
+
+Multiple contracts use the `authorized()` modifier. This modifier should be overwritten by contract to implemented the desired logic to check access to the protected methods.
+
+#### Proxy.sol
+The proxy contract implements only two functions: The constructor setting the address of the master copy and the fallback function forwarding all transactions sent to the proxy via a `DELEGATECALL` to the master copy and returning all data returned by the `DELEGATECALL`.
+
+#### DelegateConstructorProxy.sol
+This is an extension to the proxy contract that allows further initialization logic to be passed to the constructor.
+
+#### PayingProxy.sol
+This is an extension to the delegate constructor proxy contract that pays a specific amount to a target address after initialization.
+
+#### ProxyFactory.sol
+The proxy factory allows to create new proxy contracts pointing to a master copy and executing a function in the newly deployed proxy in one transaction. This additional transaction can for example execute the setup function to initialize the state of the contract.
+
+#### MasterCopy.sol
+The master copy contract defines the master copy field and has simple logic to change it. The master copy class should always be defined first if inherited.
+
+#### EtherPaymentFallback.sol
+Base contract with a fallback function to receive ether payments.
+
+#### Executor.sol
+The executor implements logic to execute calls, delegatecalls and create operations.
+
+#### ModuleManager.sol
+The module manager is an executor implementation which allows the management (add, remove) of modules. These modules can execute transactions via the module manager.
+
+A linked list is used to store the enabled modules in the smart contract. To modify the list with minimal gas usage it is required to specify the module that should be modified and the module that points to this module. This is important when disabling a module. If multiple transactions disabling modules are submitted at once it is important to note that the module that points to the module that should be disabled might have changed. The linked list requires a sentinel (start and end pointer). This sentinel is the `0x1` address. Therefore this address cannot be used as a module.
+
+#### OwnerManager.sol
+The owner manager allows the management (add, remove, replace) of owners. It also specifies a threshold that can be used for all actions that require the confirmation of a specific amount of owners.
+
+For managing the owners also a linked list (see ModuleManager.sol). Modifiying transactions that require to specify the owner pointing to the owner that should be modified include remove owner and swap owner. Also here the sentinel is the `0x1` and therefore it is not possible that this address becomes an owner.
+
+### Gnosis Safe
+#### GnosisSafe.sol
+The Gnosis Safe contract implements all basic multisignature functionality. It allows to execute Safe transactions and Safe modules.
```

Comment:
I can see only "GnosisSafe is ModuleManager, OwnerManager". Is the description deprecated?

Reply:
I think it is just how I wrote it that causes the confusion. The Gnosis Safe has all the logic to execute Safe transactions and interact with Safe module, BUT it has no logic to interact with the contract and to check if that interaction was approved by the required numbers of owners. This needs to be implemented by the sub classes.